### DISCUSSION TITLE
nvim-treesitter - [NvimTree] File system watcher failed (EMFILE) for path $path, halting watcher.

### DISCUSSION THREAD

**Discussion by emrebdr** [nvim-treesitter/nvim-treesitter#7460] [Q&A]
Hello, I'm working on a large php project, approximately ~80k files. But 2 minutes after opening this project on neovim, I get this error. 

[NvimTree] File system watcher failed (EMFILE) for path $path, halting watcher.
[NvimTree] File system watcher failed (EMFILE) for path $path, halting watcher.
[NvimTree] File system watcher failed (EMFILE) for path $path, halting watcher.
...
...
...

I get this error for all files in the project.

I don't use many plugins in neovim, only these plugins are installed => telescope, harpoon, lsp, nvim-treesitter

solutions I have tried:
```
- ulimit -n unlimited unlimited
- launchctl maxfiles unlimited unlimited
- update parser and treesitter
```

config:
```
  {
  	“nvim-treesitter/nvim-treesitter”,
  	opts = {
  		ensure_installed = {
  			“vim”, ‘lua’, ‘vimdoc’, ”php”
  		},
  	},
  }
```

Do you have any idea why this is happening or a solution? thank you 🙏

**Comment by clason** [comment]
The error comes from nvimtree, not nvim-tree-sitter (very different projects!)
